### PR TITLE
FS/PutObject: Read() data should be handled even in case of EOF.

### LINKED ISSUE #1710: browser: Object uploads file doesn't appear

### PR DESCRIPTION
**Pull request by krishnasrinivas**
Fixes #1710

### CODE REVIEW

File: fs-v1.go [hunk 1]
```diff
@@ -232,20 +232,22 @@ func (fs fsObjects) PutObject(bucket string, object string, size int64, data io.
...
-			}
-			if rErr != nil {
+			if rErr != nil && rErr != io.EOF {
 				return "", toObjectErr(rErr, bucket, object)
 			}
-			// Update md5 writer.
-			md5Writer.Write(buf[:n])
-			m, wErr := fs.storage.AppendFile(minioMetaBucket, tempObj, buf[:n])
-			if wErr != nil {
-				return "", toObjectErr(wErr, bucket, object)
```
Comment (harshavardhana) [inline fs-v1.go:245]: Needs a comment here @krishnasrinivas

File: fs-v1.go [hunk 2]
```diff
@@ -232,20 +232,22 @@ func (fs fsObjects) PutObject(bucket string, object string, size int64, data io.
 		buf := make([]byte, blockSizeV1)
```
Comment (harshavardhana) [inline fs-v1.go:234]: Add a comment here as well @krishnasrinivas

**Comment by harshavardhana** [comment]
@krishnasrinivas can you add a test case as well ?

File: fs-v1.go (hunk 1 above)
Comment (krishnasrinivas) [inline fs-v1.go:245]: @harshavardhana added comments at other places of PutObject() but any words added here will not add any value. If you have suggestions on what can be added let me know.

**Pull request merged**